tests/ConfiguratorTest.php

```
@@ -196,4 +196,40 @@ public function testParseArgsBooleanFlagsUseLastWins(): void {
 	}
+
+	public function testMergeYmlSelfInheritRecursionGuard(): void {
+		$temp_dir = sys_get_temp_dir();
+		$file     = tempnam( $temp_dir, 'wp-cli-test-self-' ) . '.yml';
+
+		file_put_contents( $file, "_:\n  inherit: " . basename( $file ) . "\nfoo: bar\n" );
+
+		$configurator = new Configurator( __DIR__ . '/../php/config-spec.php' );
+		$configurator->merge_yml( $file );
+
+		[ $config, $extra_config ] = $configurator->to_array();
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$config'. (undefined)

(UnusedLocalVariable)